### PR TITLE
feat: registration_rust

### PR DESCRIPTION
**Pull request by aldy505**

### CODE REVIEW

**Review by elianiva — APPROVED** [review]

File: registration_rust/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+[dependencies]
+axum = "0.7.5"
+tokio = { version = "1.37.0", features = ["full"] }
+tower = "0.4.13"
+redis = { version = "0.25.3", features = ["tokio-comp", "aio"] }
+serde_json = "1.0.115"
+serde = { version = "1.0.197", features = ["derive"] }
+reqwest = { version = "0.12.3", features = ["json", "gzip"] }
+chrono = "0.4.37"
```
Comment (aldy505) [inline registration_rust/Cargo.toml:18]: Might need to remove this since we don't use `chrono`

File: registration_rust/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,612 @@
...
+    calls: i64,
+}
+
+#[derive(Debug, Deserialize, Serialize, Clone)]
+struct TokenValue {
+    pub token: Option<String>,
+    pub user_email: String,
+    pub monthly_limit: i64,
+    pub revoked: bool,
+}
```
Comment (aldy505) [inline registration_rust/src/main.rs:64]: Might need to change this reflecting #172

Comment (aldy505) [reply registration_rust/src/main.rs:64]: Also, need to find a way to not render out the `token`. Probably with https://serde.rs/field-attrs.html#skip

**Review by aldy505 — COMMENTED** [review]
Might also need to run `cargo fmt` after everything's modified

**Pull request merged**